packages/client/src/internals/httpRequest.ts

```diff
@@ -59,17 +59,24 @@ export function httpRequest<TResponseShape = TRPCResponse>(
     const url = getUrl();
 
     Promise.resolve(rt.headers())
-      .then((headers) =>
-        rt.fetch(url, {
+      .then((rawHeaders) => {
+        const headers = Object.keys(rawHeaders).reduce((acc, key) => {
+          if (rawHeaders[key] !== undefined) {
+            return { ...acc, [key]: rawHeaders[key] };
+          }
+          return acc;
+        }, {});
```

Comment:
You remove most of spreads to avoid creating new object by starting with the default headers and mutating it.

Also this would probably surface the question on how to handle `string[]` values, because the type from the fetch api is `type HeadersInit = string[][] | Record<string, string> | Headers;`

```ts
        const headers: Record<string, string> = { 'content-type': 'application/json' }
        for (const key in rawHeaders) {
          if (rawHeaders[key] !== undefined) {
            headers[key] = rawHeaders[key] ;
          }
        }  
```

Reply:
Thanks for this @ArnaudBarre, I have updated this PR to use the proposed code above! Regarding the `string[]` values, I have added logic to join all array values with a `","` (this is how `fetch` natively handles `[]` header values).